documentation/ag-grid-docs/src/content/docs/server-side-model-updating-transactions/_examples/transactions-grouping/data.ts

```diff
@@ -83,7 +83,11 @@ export let data = [
 
 export function deletePortfolioOnServer(portfolio) {
     const oldDataSize = data.length;
-    data = data.filter((record) => record.portfolio !== portfolio);
+    const filteredData = data.filter((record) => record.portfolio !== portfolio);
+    // need to maintain original data reference
+    data.length = 0;
```

Comment:
I am not a fan of this syntax. What if there is a reference to the element at position higher than filteredData length? If that case in not valid then ignore this comment.

```suggestion
    A.splice(0, A.length);
```

Reply:
ok just retested. it doesn't apply and elements after filteredData.length get cleared